src/Avalonia.Controls/TreeView.cs

```diff
@@ -48,8 +48,8 @@ public class TreeView : ItemsControl, ICustomKeyboardNavigation
         /// <summary>
         /// Defines the <see cref="SelectionMode"/> property.
         /// </summary>
-        protected static readonly StyledProperty<SelectionMode> SelectionModeProperty =
-            AvaloniaProperty.Register<SelectingItemsControl, SelectionMode>(
+        public static readonly StyledProperty<SelectionMode> SelectionModeProperty =
```

Comment:
I think this should look like this:
`public static readonly new AvaloniaProperty<SelectionMode> SelectionModeProperty = SelectingItemsControl.SelectionModeProperty..AddOwner<TreeView>();`

ListBox should also have that fix.

Reply:
`SelectingItemsControl.SelectionModeProperty` is protected, would need to expose it as well then. `TreeView` does not inherit from it.

Reply:
I don't think it needs to inherit from SelectingItemsControl. I wonder why TreeView doesn't inherit from that base type.

Reply:
`TreeView` selection behavior is different as it does not use indices. To use `SelectingItemsControl.SelectionModeProperty` it must be public or internal and it will cause issues for other controls that rely on it being protected.

Reply:
Why is that the case? You can always set the SelectionMode property on all controls that inherit from SelectingItemsControl. The protected just prevents you from using `SetValue(SelectionModeProperty, value)` outside the control definition. If somone wanted to prevent this the SelectionMode property would need a protected setter. The current state just prevents you from setting the value in XAML but not in code. I guess we need a way to define read-only AvaloniaProperty to solve this in a clean way.

Reply:
I think it should be `ListBox.SelectionModeProperty.AddOwner<TreeView>();`. `SelectionMode` is defined in `SelectingItemsControl` but not made public. It's made public in derived classes (e.g. `ListBox`).

Reply:
`SelectionMode` property is protected as well, I guess for the reasons you mentioned. So for types like `Carousel` you can't set it.

Reply:
Changed implementation to use `ListBox` properties as requested.